modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk/query_processor.clj

```diff
@@ -114,12 +114,12 @@
     (u.date/parse s timezone-id)))
 
 (defmethod parse-result-of-type "DATE"
-  [_ column-mode timezone-id v]
-  (parse-value column-mode v (partial parse-timestamp-str timezone-id)))
+  [_ column-mode _timezone-id v]
+  (parse-value column-mode v (partial parse-timestamp-str nil)))
```

Comment:
Are you sure `parse-timestamp-str` can take a nil timezone?

If the value is a Unix timestamp in µs, e.g. "1.3963104E9", it will error with a null pointer exception. e.g. `(parse-timestamp-str nil "1.3963104E9")`.

Either change `parse-timestamp-str` to take a nil timezone arg, or use a different function if the value can't be a unix timestamp, i.e. `u.date/parse`.

Reply:
The call here is already guarded by the data type so I think it should be safe.

I don't think `1.3963104E9` is qualified as `date or datetime` type right?

Reply:
If that's the case, I would change this to use `u.date/parse` with one arg directly. There's no need to use `parse-timestamp-str` here.

Reply:
good call!